engine/src/conversion/codegen_rs/fun_codegen.rs

```diff
@@ -196,6 +198,16 @@ pub(super) fn gen_function(
     // which the user has declared.
     let params = unqualify_params(params);
     let ret_type = unqualify_ret_type(ret_type.into_owned());
+
+    let ret_type = if analysis.throwable {
```

Comment:
Slight preference for a single `match` statement with an arm something like `_ if !analysis.throwable => ret_type`